### PR TITLE
build: add blocklist for material unit tests

### PR DESCRIPTION
**Pull request by devversion**
Initially the blocklist has been removed because there were
no remaining disabled tests that failed. Also the blocklist
logic didn't work anymore because the `material-unit-tests` CI
job now runs against `angular/components#master` with Bazel.

388578fec946237e24521509bd887efc243a86d9 tried to revert the removal of the blocklist in favor
of a new upcoming breaking change with HammerJS, but the
revert doesn't help since the blocklist still doesn't work with Bazel.

In order to make the blocklist work with the unit tests running
with Bazel, a PR has been submitted on the components repository.
See: https://github.com/angular/components/pull/16833.

This commit updates the blocklist logic on the framework side to
work with the new logic on the components repo side.

### CODE REVIEW

**Review by josephperrott — APPROVED** [review]
LGTM

File: tools/material-ci/instructions.md [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
+
+### Excluding new tests
+
+In case there are any tests in the components test suite that fail due to
+recent changes in the framework and you want to exclude the tests temporarily,
+a new entry can be added to the `test-blocklist.ts` file.
+
+Each property in the blocklist object corresponds to a test in the components
+repository. The name of the property **must** match the exact test name. Additionally
+it's **recommended** that every entry has a field with a note on why the test is disabled.
```
Comment (kara) [inline tools/material-ci/instructions.md]: Great idea 👍

**Review by kara — APPROVED** [review]
LGTM, thanks for doing this!

**Comment by kara** [comment]
Blocked on https://github.com/angular/components/pull/16833

**Comment by josephperrott** [comment]
This PR is now unblocked as angular/components#16833 has merged

**Comment by devversion** [comment]
We still need to update the SHA for the `material-unit-tests` job as otherwise the changes made on the components repo are not taken into account. This got merged too early 😄

**Pull request closed without merging**